src/auth.rs

```diff
@@ -4,22 +4,31 @@ use actix_web::{error, Error};
 /// Checks the authorization header for the specified token.
 ///
 /// `Authorization: (type) <token>`
-pub fn check(host: &str, headers: &HeaderMap, token: Option<String>) -> Result<(), Error> {
-    if let Some(token) = token {
-        if !token.is_empty() {
-            let auth_header = headers
-                .get(AUTHORIZATION)
-                .map(|v| v.to_str().unwrap_or_default())
-                .map(|v| v.split_whitespace().last().unwrap_or_default());
-            if auth_header.unwrap_or_default() != token {
-                log::warn!(
-                    "authorization failure for {} (header: {})",
-                    host,
-                    auth_header.unwrap_or("none"),
-                );
-                return Err(error::ErrorUnauthorized("unauthorized"));
+pub fn check(host: &str, headers: &HeaderMap, tokens: Option<Vec<String>>) -> Result<(), Error> {
+    if tokens.is_some() {
+        let mut token_found = false;
+        let auth_header = headers
+            .get(AUTHORIZATION)
+            .map(|v| v.to_str().unwrap_or_default())
+            .map(|v| v.split_whitespace().last().unwrap_or_default());
+        if let Some(tokens) = tokens {
+            if !token_found {
+                for token in &tokens {
+                    if auth_header.unwrap_or_default() == token {
+                        token_found = true;
+                        break;
+                    }
+                }
             }
         }
+        if !token_found {
+            log::warn!(
+                "authorization failure for {} (header: {})",
+                host,
+                auth_header.unwrap_or("none"),
+            );
+            return Err(error::ErrorUnauthorized("unauthorized"));
+        }
     }
     Ok(())
 }
```

Comment:
I think this method can be simplified as:

```rs
pub fn check(host: &str, headers: &HeaderMap, tokens: Option<Vec<String>>) -> Result<(), Error> {
    if let Some(tokens) = tokens {
        let auth_header = headers
            .get(AUTHORIZATION)
            .map(|v| v.to_str().unwrap_or_default())
            .map(|v| v.split_whitespace().last().unwrap_or_default());
        if !tokens.iter().any(|v| v == auth_header.unwrap_or_default()) {
            log::warn!(
                "authorization failure for {} (header: {})",
                host,
                auth_header.unwrap_or("none"),
            );
            return Err(error::ErrorUnauthorized("unauthorized"));
        }
    }
    Ok(())
}
```

Reply:
Nice, let me run some tests. e.g. does the `tokens.iter().any` loop stop, when one is found. e.g. I don't want to loop over a huge array, if a token was matched in the beginning of the array already...
This is why I kept my longer construct. I thought about perf.

Reply:
Yep, the loop is discontinued as soon as a token is found. Thus this code change is awesome. Makes the code a lot shorter. Nice. Thanks for the tip.

Reply:
Done